awxkit/awxkit/cli/custom.py

```diff
@@ -182,6 +183,111 @@ class AdhocCommandStdout(HasStdout, CustomAction):
     resource = 'ad_hoc_commands'
 
 
+class AssociationMixin(object):
+
+    action = 'associate'
+
+    def add_arguments(self, parser):
+        parser.choices[self.action].add_argument('id', type=int, help='')
+        group = parser.choices[self.action].add_mutually_exclusive_group(required=True)
+        for param, endpoint in self.targets.items():
+            field, model_name = endpoint
+            if not model_name:
+                model_name = param
+            help_text = 'The ID of the {} to {}'.format(model_name, self.action)
+            group.add_argument('--{}'.format(param), metavar='', type=int, help=help_text)
+
+    def perform(self, **kwargs):
+        for k, v in kwargs.items():
+            endpoint, _ = self.targets[k]
+            try:
+                self.page.get().related[endpoint].post(
+                    {'id': v, self.action: True}
+                )
+            except NoContent:
+                # we expect to enter this block because these endpoints return
+                # HTTP 204 on success
+                pass
+            return self.page.get().related[endpoint].get()
```

Comment:
Should this return exit the loop or run after it?

Reply:
Inside it, the flags to this command are mutually exclusive:

https://github.com/ansible/awx/pull/4624/files/88dfcaa439d9cdac4a9c5de0a09771f026cf1fa6#diff-a8a950fa8ea340c107209be099b3710aR192

...so the goal is to find the first (and only) match in the list and return the results of the attach/detach of that item.